swagger_parser/lib/src/parser/utils/type_utils.dart

```diff
@@ -121,19 +125,31 @@ String? protectDefaultValue(
 }
 
 /// Protect enum items names from incorrect symbols, keywords, etc.
-Set<UniversalEnumItem> protectEnumItemsNames(Iterable<String> names) {
+Set<UniversalEnumItem> protectEnumItemsNames(Map<String, dynamic> enumMap) {
   var counter = 0;
   final items = <UniversalEnumItem>{};
+  final names = enumMap['enum'] as List<dynamic>;
 
   String uniqueEnumItemName() {
     final newName = 'undefined $counter';
     counter++;
     return newName;
   }
 
-  String numberEnumItemName(String name) {
-    final startsWithMinus = name.startsWith('-');
-    final newName = 'value ${startsWithMinus ? 'minus' : ''} $name';
+  String numberEnumItemName(dynamic enumValue) {
+    final oneOfMap = enumMap['oneOf'] as List<dynamic>?;
+    if (oneOfMap != null) {
+      final oneOfItem =
+          oneOfMap.firstWhereOrNull((e) => e['const'] == enumValue)
+              as Map<String, dynamic>?;
+
+      if (oneOfItem != null && oneOfItem['title'] != null) {
+        return oneOfItem['title'].toString();
+      }
+    }
+
+    final startsWithMinus = enumValue.toString().startsWith('-');
+    final newName = 'value ${startsWithMinus ? 'minus' : ''} $enumValue';
```

Comment:
The static analyzer flags the problems in this code

Reply:
If you fix the warning as suggested, it just gives you another warning that you haven't defined your type.  But it has to be dynamic here because it could be any type that enums support. (i.e. strings or ints or...)  Suggestions welcome.